src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -1006,8 +1006,9 @@ internal async Task Cleanup(bool async, bool connectionClosing = false, bool isD
                 Log.Debug($"Query duration time: {Connector.QueryLogStopWatch.ElapsedMilliseconds}ms", Connector.Id);
                 Connector.QueryLogStopWatch.Reset();
             }
-            Connector.EndUserAction();
+            Connector.StopCurrentActivity();
```

Comment:
Should we just do this as part of EndUserAction?

Reply:
Yeah, it makes more sense - potentially, we might want to log COPY operations and such.